rclcpp/src/rclcpp/publisher_base.cpp

```diff
@@ -259,10 +259,13 @@ void
 PublisherBase::default_incompatible_qos_callback(
   rclcpp::QOSOfferedIncompatibleQoSInfo & event) const
 {
+  const char *topic_name = rcl_publisher_get_topic_name(publisher_handle_.get());
```

Comment:
How about using its member function? And it seems not necessary to use a temporary variable.
https://github.com/ros2/rclcpp/blob/633e1157f8aeee9249afddcd0730c3f9bb894fce/rclcpp/include/rclcpp/publisher_base.hpp#L85